bmeg_app/views/app_repurpose_drug.py

```diff
@@ -0,0 +1,217 @@
+],style={'fontFamily': styles['textStyles']['type_font']})
+
+
+@app.callback(
+    dash.dependencies.Output('repurp_RESPONSE_dropdown', 'options'),
+    [dash.dependencies.Input('repurp_PROJECT_dropdown', 'value')])
+def set_options(selected_project):
+    return [{'label': k, 'value': v} for k,v in repurpose.mappings_drugResp(selected_project).items()]
+    
+@app.callback(
+    dash.dependencies.Output('repurp_RESPONSE_dropdown', 'value'),
+    [dash.dependencies.Input('repurp_RESPONSE_dropdown', 'options')])
+def set_options(available_options):
+    return available_options[0]['value']
+    
+@app.callback(
+    dash.dependencies.Output('repurp_DRUG_dropdown', 'options'),
+    [dash.dependencies.Input('repurp_PROJECT_dropdown', 'value')])
+def set_options(selected_project):
+    return [{'label': k, 'value': v} for k,v in repurpose.mappings(selected_project).items()]
+
+@app.callback(
+    dash.dependencies.Output('highlight_drugInfo', 'children'),
+    [dash.dependencies.Input('repurp_DRUG_dropdown', 'value')])
+def render_callback(DRUG):
+    q=G.query().V().hasLabel('Compound').has(gripql.eq('_data.synonym', DRUG))
+    q=q.render(['_data.synonym','_data.pubchem_id','_data.taxonomy.description'])
+    for row in q:
+        print('starting search')
+        header = row[0].upper() + ' ('+ row[1]+')'
+        descrip = row[2]
+    fig = dbc.Card(
+        dbc.CardBody(
+            [
+            html.H4(header, className="card-title",style={'fontFamily':styles['textStyles']['type_font'],'font-size' : styles['textStyles']['size_font_card'], 'fontWeight':'bold'}),
+            html.P(descrip,style={'fontFamily':styles['textStyles']['type_font'],'font-size' : '12px'}) 
+            ]),
+        color="info", inverse=True,style={'Align': 'center', 'width':'98%','fontFamily':styles['textStyles']['type_font']})
+    return fig,
+    
+@app.callback(Output("figs_repurp", "children"),
+    [Input('repurp_PROJECT_dropdown', 'value'),
+    Input('repurp_RESPONSE_dropdown', 'value'),
+    Input('repurp_DRUG_dropdown', 'value'),
+    Input('repurp_DISEASE_dropdown','value')])
+def render_age_hist(selected_project, selected_drugResp, selected_drug, selected_disease):
+    # Query
+    drugDF, disease = repurpose.get_matrix(selected_project,selected_drugResp)
+    # Preprocess:Rename drugs to common name
+    drugDF=drugDF[drugDF.columns.drop(list(drugDF.filter(regex='NO_ONTOLOGY')))] # TODO fix it so dont have to drop
+    cols=drugDF.columns
+    common=[]
+    for compound in cols:
+        q=G.query().V().hasLabel('Compound').has(gripql.eq('_gid',compound)).as_('c').render(['$c._data.synonym'])
+        for row in q:
+            common.append(row[0])
+    drugDF.columns=common
+    # Final processing + figure violins
+    finalDF,fig_violin = repurpose.compare_drugs(selected_drug, drugDF, disease,'Drug Response Metric',selected_disease )
+    finalDF.to_csv('TESTING_OUTPUT.tsv',sep='\t')
```

Comment:
Should this be in a test/fixtures directory?

Reply:
Removed unnecessary file output